ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java

```diff
@@ -171,6 +173,19 @@ public String inputFormatClassName() {
       return inputFormatClassName;
     }
 
+    public OptionalInt getBucketId() {
+      if (inputSplit instanceof PartitionAwareSplit) {
+        final PartitionAwareSplit split = (PartitionAwareSplit) inputSplit;
```

Comment:
why not move this logic into `PartitionAwareSplit`?
````
return ((PartitionAwareSplit) inputSplit).getBucketId()
````

Reply:
It seems to be a valid request. The original APIs are the result of considering future flexibility especially when we unlock this feature to non-bucketed partition transforms that might not have # of buckets. Anyway, the interface is marked as unstable, and we don't know the best API for them now
https://github.com/apache/hive/pull/5409/commits/aec8bd2a425d1886c07455d260ff3142042e9efa